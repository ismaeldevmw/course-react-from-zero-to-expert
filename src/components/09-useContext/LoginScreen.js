import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { setUser } = useContext( UserContext );

    const handleLogin = () => {
        setUser({
            id: 123,
            name: 'Ismael',
            email: 'ismael@gmail.com'
        })
    } 

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>

            <button className="btn btn-sm btn-primary" onClick={ handleLogin }>Login</button>
        </div>
    )
}
