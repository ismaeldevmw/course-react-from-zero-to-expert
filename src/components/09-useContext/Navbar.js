import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">Brand</Link>            
            <div className="collapse navbar-collapse" id="navbarText">
                <ul className="navbar-nav mr-auto">                    
                    <li className="nav-item">
                        <NavLink exact activeClassName="active" to="/" className="nav-link">Home</NavLink>                    
                    </li>
                    <li className="nav-item">
                        <NavLink exact activeClassName="active" to="/about" className="nav-link">About</NavLink>                    
                    </li>
                    <li className="nav-item">
                        <NavLink exact activeClassName="active" to="/login" className="nav-link">Login</NavLink>                    
                    </li>
                </ul>
            </div>
        </nav>
    )
}
