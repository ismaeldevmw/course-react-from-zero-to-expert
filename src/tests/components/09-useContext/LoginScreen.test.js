import { mount } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Tests in <LoginScreen />', () => {
    
    const setUser = jest.fn();

    const wrapper = mount( 
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen /> 
        </UserContext.Provider> 
    );

    test('should display correctly', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('should execute setUser with its arguments hoped', () => {
        
        wrapper.find('button').prop('onClick')();

        expect( setUser ).toHaveBeenCalledWith({
            id: 123,
            name: 'Ismael',
            email: 'ismael@gmail.com'
        });

    });
    
});
