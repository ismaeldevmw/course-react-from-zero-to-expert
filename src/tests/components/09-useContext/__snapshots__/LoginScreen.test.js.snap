// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <LoginScreen /> should display correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-sm btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
